### DISCUSSION TITLE
Welcome to Diablo2RLoader Discussions!

### DISCUSSION THREAD

**Discussion by shupershuff** [shupershuff/Diablo2RLoader#35] [Announcements]
## 👋 Yea Gidday!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas 💡
  * Engage with other community members.
  * Keep it civil will ya! 💪

Leave a comment and get involved. Cheers!

**Comment by philrea** [comment]
couple things (all good!)...

first, just want to thank you, as someone who actually is a programmer and knows there way around code at least a little bit who has recently gotten back into D2 having the greatest time the script is heading toward working wonders for me and what i can do in the game and the ways i can play it... you say anything that enhances the game is prohibited but i think, for me, you may have enhanced it as much as any bot ever couldF

second, i just read through the ps script and while i am a linux guy, unfamiliar with a lot of what goes on in the land of win shell scripting, a shell script is a shell script and they all read the same... so freaking annoying to do things you want to do without extreme verbosity and lots of having to repeat yourself over and over. on the other hand... shell scripts kick total ass because they allow you do all kinds of powerful interesting things! so not complaining just, man... wish it could have been an easier ride for you! anyways, lets get to the meat of why i am writing this right now, aside from a very heartfelt thank you...

first is the batch process which i read through and am about to go reread before it occurred to me that my question is quite simple so perhaps i should just ask and perhaps it is a common sharp edge people run up against. that is,

say i have 2 accounts (1)a1 and (2)a2. in the accounts.csv file, after correctly specifying everything in the correct format containing only 2 lines, 1 and 2 for each accounts declaration accordingly and i put a "1" in "2"s batch field: am i effectively doing the same exact thing as specifying "all" to the scripts question of "run all accounts"?

in other words, does batch mean, "use this account AND WHATEVER OTHER ACCOUNTS ARE SPECIFIED IN OTHER ACCOUNT DECLARATIONS INCLUDED IN MY 'BATCH LIST'?"

or said another way, that i think really clearifies how i am thinking about things, wouldnt the better accounts.csv in this situation look something like:
1.a1,,,,p1,,, (without any value for the batch parameter)
2.a2,,,,p2,,, (again, without any value for the batch parameter)
3.<pseudo-account-1>,,,,<pseudo-password-1>,"1,2",,,

where the logic goes, declare both accounts independently of each other first. then, if the response to what account i wish to run is returned "3" then what is desired is, to "batch both accounts 1 and 2" where i am interpreting "batch" as, "open multiple accounts according to what is specified" which in this case would be "whatever account 1 decalres" and whatever "account 2 declares"... since they only declare themselves, only those 2 accounts should be launced...

but it does not seem thats how it works because there are no "pseudo" or "placeholder" accounts to my knowledge. which means that in the second account if i included "1" as the batch parameter value the logic it would be indicating is to - first open me, number 2... then go check out whatever is going on in account "1" and do as it says as far as launching d2r and accounts...

is my understanding correct? if it is then a2,pw2,"1" under account "2" would indicate launch me and "1", which implies... from the loader there is no way to launch the second account alone...

if this is the case, good fine, your tool is for multiple account launches, it need not serve as the universal d2r launch hub. however, if you were to clarify on those points perhaps it woudl be easier for those wanting to use the tool to get the most fundamental use of that tool?

honest to god, not a criticism, you script has worked decently for me thus far but i have been running into some issues and thought hmm, perhaps its because im not uderstanding what "batch" means...

truth be told im just gonna hit up the "all" option most of the time but in trying to figure what is going on behind the seens with windows and what not, this is a point of confusion right now! and, in all honesty things got so shaky with however i managed to f it all up that i got worried about my bnet accounts and system stability, that i decided to head here! anyways what follows is a description of the very odd thing going on earlier hoping someone can shed light on the topic and then my 2 cents on the more important parts of the tool - a few ideas for quotes you might want to include in your quotes list and a special message from one human to the next at the very end!

k, so what got me all hot and bothered earlier inspiring me to write this message is that i was running 2 instances of D2R on 2 accounts for the simple purpose osetting up some mules on a second account as well as xferring old ladder gear onto the second account mules. i did this buy letting both instances operate at full screen while using alt-tab. the weird thing that kept happening, over and over and over again was that, when i would alt-tab back n forth from mule to old char/mule, probably 1 out of every 2 or 3 times - my other accountsd character (the not in focus one) would drop from the game and show me the "connection interuptted" message. were talkin this happened a lot, over and over and over again the entire time... but from what i recall only once or twice was i dropped from bnet totally most of the time i was just kicked out of game and could just immediately join again...

anyone have any clue what might be going on here? (my guess was that your tool was not the culprit but something like 2 different accounts, same IP, same game, same server, hell, almost the same damn socket if it wer possible - was doing either weird or something blizzard doesnt like or considers an indicator of error?) anyways, anyone anything? thanks for whatever you got seriously! on to the more pressing matters...

as for quotes i want you to add to your list...

1. obviously, if you have "oh no snakes, i hate snakes..." you HAVE to have "'IM SICK AND TIRED OF THESE MOTHER 'N SNAKES ON <<THESE>> MOTHER -N <<PLAINS>>' - attributed to what is believed to be a powerful paladin as he moved across the Plains of Despair"...

ya know, cause Samuel L is a black guy... hence paladin! and regardless of if there are any areas in the game for which you can validly call "plains" with snakes in em? ahhh who cares, and who knows, mayber there were when SLJackson was perusing the area back in the day... you dont know...

and the second and far better one would be...

2. inspired by your quote depicting a happy go lucky caring Andariel in the past i couldnt help but start spinning my wheels over izual...

ill be the first to admit... sometimes i get an idea in my head started from almost nothing and then just go on and on and on with it. here the thought has tempted me to write Diablo lore with a oppositional them to it "from hell's point of view" portraying our side as the actual evil ones the whole time... anyways i digress, where was i? oh yeah a quote... or at least an idea for what i think would be extremely funny if you could figure out the right way to convey it! something like,

"To my eternal brother Izual whose strength and morals, heart and compassion, and yet wise and decent are such that I have seen no equal. The love i have for my little brother runs deeper than my Soul" - the words of Archangel Tyriel engraved upon his very own soulstone which he gave as a wedding present to prove his loyalty to his unbreakable, uncompromising princpled and moral man that was his brother Izual. (approx. 6 weeks before izual caught his newly married wife with that weird kid that seemed to love cows so much somewhere along the road to Tristram...

yeah ok not great but cmon... quotes that represent the burning hells as merely misunderstood and the so-calledd light or good as the true evil can always be funny and something about Izual being as solid and moral as possible before finding his wife f-king that boy Wirt from Tristram. and having that be the trigger that made him sell out all of the entirety of life in the universe...?

it shits all over everything in Diablo/Sanctuary Lore so hard that stuff like it need to be put somewhere! also, something a long the lines of Tyriel knowing about the affair or actively partaking in it in some way are more fun to add to the new lore!

yeah yeah. i rant... f it you should too its fun!

**Reply by philrea** [reply]
oh, almost forgot. a message from human to human. the next project you work on, if you arent sure, use the following color scheme or some variation of it: #e4cbab #f9f7f5 #858483 #dbe4bd

its for my dog lucky. hes still around and still the best friend i have ever and will ever have for my entire existence but, hes getting older and i cant seem to stop it...

he is my Toby... to the ones we love my friend! to the ones we love...

**Comment by zamundanet** [comment]
Hello shupershuff. Thank you so much for this loader. I tried to setup everything but the game does not seem to connect me to the online servers.

**Reply by shupershuff** [reply]
Hi, sorry just seen this. Did you get it sorted?

**Comment by sh4nks** [comment]
Awesome loader, it was exactly what I was looking for. I love the simplistic nature of it being just a CLI UI. Before I just had my own little script which just killed the D2R handles. Thanks!

Next step would be to have something like this for Lutris/Proton :D

**Reply by shupershuff** [reply]
Glad you enjoy it :)

**Comment by diabloinnaset-stack** [comment]
i try to open the drloader with power shelll but i doesnt do anything

**Reply by shupershuff** [reply]
Probably a case of an instruction missed.
Checkout [this section](https://github.com/shupershuff/Diablo2RLoader/?tab=readme-ov-file#1-download) and [this section](https://github.com/shupershuff/Diablo2RLoader/?tab=readme-ov-file#4-run-the-script-manually-for-the-first-time).

Let me know how you get on.

**Comment by MrViviD2R** [comment]
Hello!
I would like to say HUGE THANK YOU my friend!
I am using your loader for a very long time and just because I am a github newb I did not know how to reach any comment section nor how to say thank you :D 

There is however one Issue I would like to addreess to make this loader perfect (for my usecase).
When I recently added 8th account to my list (now I can proudly fill the entire game) I realised that no matter how I fiddle around with tokens I cannot make it work. I can't see any way out, I tried setting up the entire loader again from scratch and it always works up to 7 accounts. The only way to run 8 accs simultaneously i to run 7 of them using the loader and the las one either through the battle.net launcher or using a dedicated shortcut with launch parameters. 

The error message is "cannot connect to server" and it always happens for this very specific account, no matter how many I have opened before or after. The game loads straight into the "offline" tab and there's nothing I can do about it.

**Reply by shupershuff** [reply]
Gday mate, thanks for the kind words!

What auth method are you using for the 8th account?
If using parameter (username and password) the most obvious cause is a typo in the email address or password. Also, with this method, MFA enabled accounts annoyingly don't work either.

There have been some issues since early December with token auth (blizz issuing tokens that expire after 30min) but a few folk have told me that this isn't an issue anymore.

**Reply by MrViviD2R** [reply]
Im using tokens and everything works rock solid (7 accounts). I tried obtaining a new token for the 8th account a several times and it didnt help. I mean it helped only for the first launch. When I restart the launcher and lauch the 8 instancies again, the 8th will fail to connect until I obtain a new token.

**Reply by MrViviD2R** [reply]
@shupershuff so I guess no solution for me?

**Reply by shupershuff** [reply]
Hey @MrViviD2R, have you tried getting a new token for the impacted account in the last week?

**Reply by MrViviD2R** [reply]
> Hey @MrViviD2R, have you tried getting a new token for the impacted account in the last week?

I think it finally works right now. :)

**Reply by shupershuff** [reply]
Woohoo! :)

**Comment by voidwardz** [comment]
hi @shupershuff 

Just wanted to report that I've gotten my account suspended for "suspicious activity" when using your launcher. It started just now, after I tried switching to EU for a trade. I tried to relaunch with the script on NA as well and the message persists. However, when using the regular bnet launcher, everything goes fine. Other than your script, I'm only using a popular loot filter, which I've never had any issues with. Not the end of the world since I can connect normally, but I thought I'd let you know. I'll update this if the lock clears after a while...

**Reply by MrViviD2R** [reply]
don't freak out, I'm using this loader for a veeery long time and I experienced this issue you mentioned many many times. It's caused sometimes by some token issues, overlaping of instancies or / and too quick realm / region changes. It always goes away after I close every single d2r process and the launcher itself and run it again. It's nothing to worry about. Sometimes you need to launch the bnet client to update the game itself, sometimes its something else, but it's a regular thing that occurs, and you have to get over it not freaking out :)

**Reply by shupershuff** [reply]
Only some accounts are impacted. If this happens to you, use Token method for authentication instead of Parameter.
See [instructions](https://github.com/shupershuff/Diablo2RLoader?tab=readme-ov-file#2-setup-your-accounts) in the readme which outlines this.

**Reply by voidwardz** [reply]
Thanks for the quick response guys, much appreciated. Love the launcher other than this :) I'll look into Token stuff if it persists.

**Comment by frakira** [comment]
Hi,

d2emu is already updated for clone stages for the new game mode.. Do you plan to update your script?
(I really love your script and I am recommending it to everybody)

**Reply by shupershuff** [reply]
I hope to have an update out by the end of the week

**Comment by Zoxjib** [comment]
I understand Steam support is not planned, and to be honest I'm not sure it's even possible at all to launch multiple game copies from different Steam accounts.

However, if we had the option to at least launch 1 copy from Steam (basically run the game .exe from the Steam folder), it would allow us to utilize the script to conveniently launch from the script and utilize the great features like Settings.json automatic replacement, selecting  Region, etc.

I believe many probably have 1 copy on Steam and all of their alts on Battle.net.

**Reply by shupershuff** [reply]
I doubt many people at all have Steam accounts given the game has only been available via steam for a week.
I would support it if I could, I just don't have any information about how it authenticates and I presume it's quite complicated.

For my own curiosity, I'm keen to see what launch parameters the steam version uses.

Can you open the game (normally through steam), then open task manager (Ctrl + Shift + Esc), then click on the details tab to see the list of processes.
Then, can you right click on any of the columns and choose the option for 'select columns'
From here, scroll down to find the 'command line' option and add that.
From here find the d2r process.
Can you take a screenshot (Windows Key + Shift + S) of what appears in the command line column for this process?

It should look something like this:
<img width="681" height="126" alt="image" src="https://github.com/user-attachments/assets/782563a7-6b00-4baa-a896-e1dc9cd97b37" />

**Comment by Zoxjib** [comment]
I think you'd be surprised, the game is doing really well on Steam. People love having all their games on one platform, and Steam is the one for the majority. Hours tracking, overlay, achievements and more are some reasons why people would launch their Main client through Steam instead of Battlenet.

<img width="265" height="131" alt="firefox_czmFDJ58RP" src="https://github.com/user-attachments/assets/d7117b6a-6ec3-4bd4-b089-d891033468af" />
<img width="957" height="172" alt="firefox_hmrkurY1DB" src="https://github.com/user-attachments/assets/32bde1a1-9728-48f6-b771-808c37ce5a26" />


Here's my Command line (I added all of those things manually in Steam's Launch Parameters)
Without my modification, it would've read just "C:\Program Files (x86)\Steam\steamapps\common\Diablo II Resurrected\D2R.exe"

<img width="1266" height="28" alt="fLY6mc0HSE" src="https://github.com/user-attachments/assets/054a03c6-512b-4829-9d47-e3815209c0cf" />

**Reply by shupershuff** [reply]
I sit corrected, thanks.

Very odd that the launch params are different.

**Reply by shupershuff** [reply]
Can you please confirm if changing the parameter for -address us.actual.battle.net actually works?
EG If you change it to kr.actual.battle.net does it direct you to asia instead.

**Reply by Zoxjib** [reply]
As I understood, all Blizzard games that are launched through Steam can only be (and must be) linked to a single Battlenet account. You do it once when you login into a Blizzard game on Steam, and then it appears as a Connection under your Battlenet account. At that point, Steam just launches the .exe and it automatically uses the Connected Battlenet account. I just know that many like me would love launching a main account through Steam.

Yes, the -address definitely works, that seems to be the only way to change regions when playing through Steam (there's no button in the UI to choose a region).

**Reply by shupershuff** [reply]
So just to be super clear, when you open D2r on steam for the first time it forces you to make a battle.net account or sign into one?
Does that mean when you open the battle net client with that account you see D2r as an option?
If so then my script will already work for accounts purchased via Steam.

Also, can you please open regedit.msc and navigate to this path:
Computer\HKEY_CURRENT_USER\SOFTWARE\Blizzard Entertainment\Battle.net\Launch Options\

And let me know if there's a key for URI_Token? I don't need the data, just want to know if it exists.

**Reply by Zoxjib** [reply]
Regarding sign in - yes, precisely. In my case I've done that about a year ago with Diablo IV. When I added D2R last week, it automatically knew to use the same account.

Regarding content on Battlenet vs. Steam, not exactly... I see D2R on that account because I purchased it on both platforms. If you only purchased it on Steam, it won't appear in Battlenet and you'll only be able to launch in through Steam. 

In my case, I haven't bought the DLC on Battlenet, but I do own the base game, so when I launch the game on Battlenet I can see my Warlock that I levelled through Steam, but the Play icon is disabled because I don't own the content on Battlenet. Wild 🤭

The URI_TOKEN exists.

**Reply by shupershuff** [reply]
So you've bought it on Steam and for the battlenet account you've linked it too, you've also been able to buy it again using that very same account?

If so that's completely scuffed (on blizzards part).

I was hoping to simply get you to authenticate with your battle.net details.
Have you tried launching d2r with this account via my launcher?

**Reply by Zoxjib** [reply]
Yep, you got it exactly right. Blizzard are double-dipping, selling 2 copies of the same game on different platform. They've been doing it with D4, CoD, etc.

I tried launching my main account through your launcher and it behaved the same as launching the game through the Battlenet launcher - DLC is disabled because I haven't purchased it there, and I have none of Steam's benefits.

**Reply by shupershuff** [reply]
I could _maybe_ add a thing for you to be able to launch 1 (and only 1) account from Steam by selecting auth method as "Steam" in the future using steam://run/2536520.

It more than likely won't be in tonights/tomorrows update though.

**Reply by shupershuff** [reply]
@Zoxjib Can you please send me a video (eg Loom.com) of what the sign in process looks like from opening the game on steam to getting to the character select screen?

**Reply by Zoxjib** [reply]
In about an hour and a half, sure. But to sum it up - there is no sign-in process. The game just launches straight to character select menu, regardless of if Battlenet is on or not.

**Reply by shupershuff** [reply]
In that case, don't worry about the video. I just wanted to ensure that the behaviour described is what happens.

**Reply by Zoxjib** [reply]
Sure... but here it is anyway so you have a reference.

https://github.com/user-attachments/assets/912dfeb1-38de-463f-bf19-852810e65955

**Reply by shupershuff** [reply]
@Zoxjib wanna be my test dummy for this mate?
Download this file, overwrite your current script with it and let me know if it works or not:
https://github.com/shupershuff/Diablo2RLoader/blob/Steam/D2Loader.ps1

Once downloaded, add an entry in accounts.csv. You can leave email, password and token fields blank, all you need to do is enter "Steam" in the authentication method column.

**Reply by Zoxjib** [reply]
When I try to launch the new account, because the Password column for that account is empty, the script is requesting to input a Battlenet password

<img width="565" height="69" alt="image" src="https://github.com/user-attachments/assets/1488350e-db86-4244-82ec-b693d03bf47c" />

**Reply by shupershuff** [reply]
It shouldn't need a password.
Is the authentication method set to steam for that account in accounts.csv?

**Reply by Zoxjib** [reply]
Yes, here are the redacted contents of my Accounts.csv

<img width="1171" height="216" alt="NcgG7V1nWf" src="https://github.com/user-attachments/assets/1bd93c6d-0e48-4c37-98e3-915f90a6c286" />

**Reply by shupershuff** [reply]
Cheers. Looks good, I'll have a look in a couple days when I'm next home.

**Reply by Zoxjib** [reply]
I tried launching the game by selecting the Batch rather than through the account ID, and it worked!
However it launched with additional parameters that are not necessary for Steam and issued a warning I haven't seen before:

<img width="631" height="299" alt="steamwebhelper_YfxfV8uQqn" src="https://github.com/user-attachments/assets/6946ca87-c3e4-49fa-b2d6-bc93f6a675ae" />

I could add "-direct -txt -resetofflinemaps -enablerespec" to the Steam Launch params and remove them from Accounts.csv, but it would still try to add -uid osi --instance1

I don't know if -uid osi --instance1 does anything on Steam.

**Reply by shupershuff** [reply]
I removed -uid osi and fixed it prompting for battlenet details.

Try the updated steam test build from here: https://github.com/shupershuff/Diablo2RLoader/blob/Steam/D2Loader.ps1

**Reply by voidwardz** [reply]
It works but I get the same warning as the above user -- except for me it said --instance3 only (I only have 3 accounts). Script says "D2r didn't launch as expected" and doesn't seem to detect that Steam/D2R is in fact open (still shows green instead of Active).

**Reply by shupershuff** [reply]
Thanks. Not sure if there's a way to avoid that warning. The flag --instance3 is used by the script to dermine which d2r process relates to which account.

Can you both please send me a screenshot of what you see in task manager's command line column as per [these instructions](https://github.com/shupershuff/Diablo2RLoader/discussions/35#discussioncomment-15858742).

**Reply by Zoxjib** [reply]
Works through Batch or ID now, awesome!. As noted above, Steam warns about the custom command line due to the "--instance 1" argument

<img width="749" height="27" alt="image" src="https://github.com/user-attachments/assets/ec53be32-7a27-4d3e-907d-7b5858c13f25" />

**Reply by shupershuff** [reply]
Fixed the "d2r didn't launch as expected" issue in the steam branch. Forgot to remove out a statement I left in there for testing haha.

**Reply by shupershuff** [reply]
Thanks @Zoxjib. Looks like the arguments are being added on as expected which is good. I just tried launching Age of Empires in my steam library using a launch command and I get the same warning. After a quick, 5min look online I couldn't find a way to disable this warning unfortunately.

**Reply by Zoxjib** [reply]
It's a valid warning but in the grand scheme of things, it's a non-issue. I'm very happy with the feature.
Thank you!

**Reply by voidwardz** [reply]
@shupershuff This update still only enables one Steam account, correct?

**Reply by sh4nks** [reply]
@shupershuff I guess launching using ``steam -applaunch <gameid> <params>`` will also add this warning/popup?

**Reply by shupershuff** [reply]
> @shupershuff I guess launching using `steam -applaunch <gameid> <params>` will also add this warning/popup?

In quick testing with this seems to be a better option as i don't see the pop up. Will need to add some steps to find steam.exe file path though.

**Reply by shupershuff** [reply]
> @shupershuff This update still only enables one Steam account, correct?

For now yes. It may be possible to launch separate Steam accounts using other parameters (-steamuser & -steampass) but I haven't tested that yet or seen what the experience is like.

**Reply by shupershuff** [reply]
Alrighty team, [give this a hoon](https://github.com/shupershuff/Diablo2RLoader/blob/Steam/D2Loader.ps1) and let me know if this is all good.

**Reply by Zoxjib** [reply]
It failed to launch through Steam. The first attempt the script echoed 'D2r isn't open yet for 1. Waiting another second...'
On the second attempt it failed and Steam popped up the following window:
<img width="470" height="187" alt="steamwebhelper_5pPEAcRSmm" src="https://github.com/user-attachments/assets/10baefc6-7287-4405-bc92-7b01ffa4b3f9" />

**Reply by shupershuff** [reply]
Try again. I accidentally left the age of empires game ID in there instead of the D2r one haha

**Reply by Zoxjib** [reply]
Now it worked and the game launched with all the parameters and without any warning popups. 👏

**Reply by shupershuff** [reply]
> It works but I get the same warning as the above user -- except for me it said --instance3 only (I only have 3 accounts). Script says "D2r didn't launch as expected" and doesn't seem to detect that Steam/D2R is in fact open (still shows green instead of Active).

@voidwardz Please advise if this is still an issue. If so please advise what your game install path is set to and if this is different from the steam install path.

**Reply by voidwardz** [reply]
@shupershuff all good now! ty so much.

**Reply by Zoxjib** [reply]
Looks like the d2emu pause was fixed too. The error is visible but doesn't pause the script

<img width="660" height="61" alt="powershell_0gZ3pCYvlS" src="https://github.com/user-attachments/assets/bb8350b2-aa29-4cb8-aba6-6908727c63c3" />

**Reply by shupershuff** [reply]
> Looks like the d2emu pause was fixed too. The error is visible but doesn't pause the script
> 
> <img alt="powershell_0gZ3pCYvlS" width="660" height="61" src="https://private-user-images.githubusercontent.com/10631966/556303935-bb8350b2-aa29-4cb8-aba6-6908727c63c3.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzIyNzM1ODcsIm5iZiI6MTc3MjI3MzI4NywicGF0aCI6Ii8xMDYzMTk2Ni81NTYzMDM5MzUtYmI4MzUwYjItYWEyOS00Y2I4LWFiYTYtNjkwODcyN2M2M2MzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjI4VDEwMDgwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhZDcwYjE5ZjFjNDZiMjUzMDM0MDZlZDg4OWIyMTgyN2JjYmEzN2VhMGUyNjMyMTNlMDRhMGUxNmQ2NzM5NzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SesOg1lG8cgpmCKKMhQyfrCoCOJTR8OJf2K3MHtxEaY">

Yep I removed the pause but kept in the error. Can't help ya with the connectivity woes sorry!

**Reply by shupershuff** [reply]
Feel free to try the [latest test version for steam](https://github.com/shupershuff/Diablo2RLoader/blob/Steam/D2Loader.ps1). Has some minor improvements for the steam side of things but also introduces a couple of new features (TZ overlay and Shortcut keys for switching windows).

**Reply by voidwardz** [reply]
@shupershuff That's great, will test! But I think you forgot to add the option for window switching in the config.xml for the Steam branch, I only see it on the main one :)

**Reply by voidwardz** [reply]
I've reverted back to autohotkey for window switching; yours worked but there was a slight lag between switching from 1 to 3, 4 and 5, but it was instantaneous for the back and forth between 1 and 2. Autohotkey has no delay for me with all of them. The TZ overlay works well! I'd suggest having an option to only have the next TZ visible though, since we have the current one by default in the game already.

**Reply by shupershuff** [reply]
> @shupershuff That's great, will test! But I think you forgot to add the option for window switching in the config.xml for the Steam branch, I only see it on the main one :)

I only updated d2loader.ps1. download that and overwrite your current one.

**Reply by shupershuff** [reply]
> I've reverted back to autohotkey for window switching; yours worked but there was a slight lag between switching from 1 to 3, 4 and 5, but it was instantaneous for the back and forth between 1 and 2. Autohotkey has no delay for me with all of them. The TZ overlay works well! I'd suggest having an option to only have the next TZ visible though, since we have the current one by default in the game already.

I'll investigate making switching a bit snappier.

In terms of the TZ overlay, originally I had it displaying next TZ only, however I found it a bit impractical as more often than not I'm in shared games where it's not easy to see what the current TZ is.

**Reply by shupershuff** [reply]
I have uploaded another version to the Steam branch to enable opening multiple steam accounts. Unfortunately I can't test this as I don't have D2r on Steam.
The experience for multiple accounts is pretty poor as it doesn't seem possible to pass username and password parameters to steam.exe, as such you have to manually sign in for each account. Gross.

To test, download the [latest copy of D2Loader.ps1 from the Steam Branch](https://github.com/shupershuff/Diablo2RLoader/blob/Steam/D2Loader.ps1) and copy this over the top of your existing D2Loader.ps1 file to overwrite it.

Keen to see if this works with DisableIconStacking set to True/False.

**Comment by Zoxjib** [comment]
It looks like Steam receives updates faster/earlier too. 
Steam D2R client just updated to 3.1.91735, while BNet is still on 3.1.91636

**Comment by voidwardz** [comment]
@shupershuff it works fine now with the latest update :) no warning.
